lib/unstable/Config/Lexicon/ConfigLexiconEntry.php

```diff
@@ -47,12 +48,37 @@ public function __construct(
 		private readonly ?string $rename = null,
 		private readonly int $options = 0,
 	) {
+		// key can only contain alphanumeric chars and _-.
+		if (preg_match("/[^[:alnum:]\-._]/", $key)) {
```

Comment:
```suggestion
		if (preg_match("/[^[:alnum:]_]/", $key)) {
```

But I'd say this can go separate later after the documentation stuff is done and merged.